libs/ledger-live-common/src/families/tezos/signOperation.ts

```diff
@@ -32,53 +33,101 @@ export const signOperation: SignOperationFnSignature<Transaction> = ({
           };
 
+          const contents: OperationContents[] = [];
+
+          if (!(account as TezosAccount).tezosResources.revealed) {
+            const revealFees = await tezos.estimate.reveal();
+
+            contents.push({
+              kind: OpKind.REVEAL,
+              fee: DEFAULT_FEE.REVEAL.toString(),
+              gas_limit: (revealFees?.gasLimit || 0).toString(),
+              storage_limit: (revealFees?.storageLimit || 0).toString(),
+              source: publicKeyHash,
+              counter: (Number(sourceData.counter) + 1).toString(),
+              public_key: publicKey,
+            });
+          }
+
           switch (transaction.mode) {
-            case "send":
-              res = await tezos.contract.transfer({
-                mutez: true,
-                to: transaction.recipient,
-                amount: transaction.amount.toNumber(),
-                ...params,
+            case "send": {
+              type = "OUT";
+
+              contents.push({
+                kind: OpKind.TRANSACTION,
+                amount: transaction.amount.toString(),
+                destination: transaction.recipient,
+                source: freshAddress,
+                counter: (Number(sourceData.counter) + 1 + contents.length).toString(),
+                ...transactionFees,
+              });
+
+              forgedBytes = await rpc.forgeOperations({
+                branch: block.hash,
+                contents,
               });
-              opbytes = res.raw.opbytes;
+
               break;
-            case "delegate":
-              res = await tezos.contract.setDelegate({
-                ...params,
+            }
+            case "delegate": {
+              type = "DELEGATE";
+
+              contents.push({
+                kind: OpKind.DELEGATION,
                 source: freshAddress,
+                counter: (Number(sourceData.counter) + 1 + contents.length).toString(),
                 delegate: transaction.recipient,
+                ...transactionFees,
               });
-              opbytes = res.raw.opbytes;
-              type = "DELEGATE";
+
+              forgedBytes = await rpc.forgeOperations({
+                branch: block.hash,
+                contents,
+              });
+
               break;
-            case "undelegate":
-              res = await tezos.contract.setDelegate({
-                ...params,
+            }
+            case "undelegate": {
+              type = "UNDELEGATE";
+
+              // we undelegate as there's no "delegate" field
+              // OpKind is still "DELEGATION"
+              contents.push({
+                kind: OpKind.DELEGATION,
                 source: freshAddress,
+                counter: (Number(sourceData.counter) + 1 + contents.length).toString(),
+                ...transactionFees,
               });
-              opbytes = res.raw.opbytes;
-              type = "UNDELEGATE";
+
+              forgedBytes = await rpc.forgeOperations({
```

Comment:
We could technically use rpc.forgeOperation after all switch cases as we are always providing the same parameters